### PR TITLE
use BPMN errorMessageVariable

### LINKED ISSUE #30: Create own service for SearchBundle testing instead of testing as part of service delegate
Refactor the testing of a SearchBundle out of the ServiceDelegate in its own service class in order not to have the setup for the whole delegate with Mockito for testing.

### LINKED ISSUE #42: Use BPMN errorMessageVariable instead of specific self set variable
Use the [camunda:errorMessageVariable](https://docs.camunda.org/manual/7.22/reference/bpmn20/events/error-events/#defining-an-error) extension in the BPMN instead of setting additional variables like`ConstantsReport.BPMN_EXECUTION_VARIABLE_REPORT_RECEIVE_ERROR_MESSAGE`.

### PR DESCRIPTION
**Pull request by wetret**
resolves #30
resolves #42

### CODE REVIEW

File: src/main/java/de/medizininformatik_initiative/process/report/util/SearchQueryCheckService.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+			.map(ResourceType::name).collect(Collectors.toSet());
+
+	private static final List<String> DATE_SEARCH_PARAMS = List.of("date", "recorded-date", "onset-date", "effective",
+			"effective-time", "authored", "collected", "issued", "period", "location-period", "occurrence");
+	private static final List<String> TOKEN_SEARCH_PARAMS = List.of("code", "ingredient-code", "type");
+	private static final List<String> OTHER_SEARCH_PARAMS = List.of("_profile", "_summary");
+	private static final List<String> VALID_SEARCH_PARAMS = Stream
+			.of(DATE_SEARCH_PARAMS.stream(), TOKEN_SEARCH_PARAMS.stream(), OTHER_SEARCH_PARAMS.stream()).flatMap(s -> s)
+			.toList();
+
```
Comment (hhund) [inline src/main/java/de/medizininformatik_initiative/process/report/util/SearchQueryCheckService.java:39]: empty line

File: src/main/java/de/medizininformatik_initiative/process/report/spring/config/ReportConfig.java [hunk 2]
```diff
@@ -94,7 +95,14 @@ public DownloadSearchBundle downloadSearchBundle()
 	@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
 	public CheckSearchBundle checkSearchBundle()
 	{
-		return new CheckSearchBundle(api);
+		return new CheckSearchBundle(api, searchQueryCheckService());
+	}
+
+	@Bean
+	@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
```
Comment (hhund) [inline src/main/java/de/medizininformatik_initiative/process/report/spring/config/ReportConfig.java]:
```suggestion
```

Is prototype scope needed? Service seems to have no persistent state. A singleton instance should be fine.

**Review by hhund — APPROVED** [review]

File: src/main/java/de/medizininformatik_initiative/process/report/util/SearchQueryCheckService.java (hunk 1 above)
Comment (wetret) [reply src/main/java/de/medizininformatik_initiative/process/report/util/SearchQueryCheckService.java:39]: removed with b72664366a0b71bf85d886b303618e96ee5760e2

File: src/main/java/de/medizininformatik_initiative/process/report/spring/config/ReportConfig.java (hunk 2 above)
Comment (wetret) [reply src/main/java/de/medizininformatik_initiative/process/report/spring/config/ReportConfig.java]: changed with b72664366a0b71bf85d886b303618e96ee5760e2

**Pull request merged**